cmd/broker.go

```diff
@@ -0,0 +1,211 @@
+//
+// Copyright (c) 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/automationbroker/bundle-lib/clients"
+	"github.com/automationbroker/sbcli/pkg/util"
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type bootstrapResponse struct {
+	SpecCount  int `json:"spec_count"`
+	ImageCount int `json:"image_count"`
+}
+
+var brokerName string
+
+var brokerCmd = &cobra.Command{
+	Use:   "broker",
+	Short: "Interact with an Automation Broker instance",
+	Long:  `List or Bootstrap bundles on an Automation Broker instance`,
+}
+
+var brokerCatalogCmd = &cobra.Command{
+	Use:   "catalog",
+	Short: "List available service bundles in broker catalog",
+	Long:  `Fetch list of service bundles in Automation Broker catalog`,
+	Run: func(cmd *cobra.Command, args []string) {
+		listBrokerCatalog()
+	},
+}
+
+var brokerBootstrapCmd = &cobra.Command{
+	Use:   "bootstrap",
+	Short: "Bootstrap an Automation Broker instance",
+	Long:  `Refresh list of bootstrapped service bundles in Automation Broker catalog`,
+	Run: func(cmd *cobra.Command, args []string) {
+		bootstrapBroker()
+	},
+}
+
+func init() {
+	brokerCmd.PersistentFlags().StringVarP(&brokerName, "name", "n", "openshift-automation-service-broker", "Name of Automation Broker instance")
```

Comment:
Is this the namespace of the broker, or something else?

Reply:
Name of the `clusterservicebroker` resource which we expect to match the namespace the broker lives in. I'm not opposed to adding more flags to allow users to use this with brokers that aren't installed in the manner we expect, but for first phase I'd like to keep it drop dead simple.